### DISCUSSION TITLE
OpenSSL installation seems broken (docker)

### DISCUSSION THREAD

**Discussion by esatterwhite** [Kong/kong#8024] [Help] [answered]
I'm trying to test a couple of things with the jwt parser that ships with kong, but when I try to require in it a lua repl it complains about missing symbols from the open ssl install
```shell
LuaJIT 2.1.0-beta3 -- Copyright (C) 2005-2021 Mike Pall. https://luajit.org/
JIT: ON SSE3 SSE4.1 fold cse dce fwd dse narrow loop abc sink fuse
> require 'kong.plugins.jwt.jwt_parser'
/usr/local/share/lua/5.1/resty/openssl/version.lua:60: OpenSSL has encountered an error: /usr/local/share/lua/5.1/resty/openssl/version.lua:45: Symbol not found: SSLeay
stack traceback:
        [C]: in function 'error'
        /usr/local/share/lua/5.1/resty/openssl/version.lua:60: in main chunk
        [C]: in function 'require'
        /usr/local/share/lua/5.1/resty/openssl/include/evp.lua:7: in main chunk
        [C]: in function 'require'
        /usr/local/share/lua/5.1/resty/openssl/digest.lua:6: in main chunk
        [C]: in function 'require'
        /usr/local/share/lua/5.1/kong/plugins/jwt/jwt_parser.lua:9: in main chunk
        [C]: in function 'require'
        stdin:1: in main chunk
        [C]: at 0x55f267616460
```
I'm running this in the 2.6.0-alpine docker image
is there something I need to do to make that work? or is there something broken with the install?

**Comment by fffonion** [comment, marked as answer]
@esatterwhite we assume openssl is always linked in the process, try using `resty` instead of the `luajit` repl.

**Reply by esatterwhite** [reply]
Is there a repl for resty that I'm just not aware of that still work?

**Reply by fffonion** [reply]
You can use https://github.com/saks/lua-resty-repl, and then `resty -e "require('resty.repl').start()"`

**Comment by esatterwhite** [comment]
I think this might be broken from a package standpoint. Today was the day github started decommissioning legacy git protocols

```txl
luarocks install lua-resty-console --local
Installing https://luarocks.org/lua-resty-console-0.1.1-1.src.rock
Missing dependencies for lua-resty-console 0.1.1-1:
   lua-hiredis-with-5.2-fix >= 0.2.2 (not installed)
   inspect >= 2.0 (not installed)
   argparse >= 0.6.0 (not installed)

lua-resty-console 0.1.1-1 depends on lua-hiredis-with-5.2-fix >= 0.2.2 (not installed)
Installing https://luarocks.org/lua-hiredis-with-5.2-fix-0.2.2-1.rockspec
Cloning into 'lua-hiredis'...
fatal: remote error: 
  The unauthenticated git protocol on port 9418 is no longer supported.
Please see https://github.blog/2021-09-01-improving-git-protocol-security-github/ for more information.
```

It kind of seems like this breaks luarocks in general.

**Reply by fffonion** [reply]
Is lua-resty-console an implicit dependency for Kong or the plugin you are working on? I think you will need to ask the package maintainer (https://github.com/nicoster/lua-hiredis i assume) to release a new rockspec,
all Kong dependency rocksepc uses git+https protocol from I remember.

**Reply by esatterwhite** [reply]
I would like to have a repl for development. Not specifically a requirement for the functionality of the plugin
